straxen/plugins/event_patternfit.py

```diff
@@ -478,22 +479,30 @@ def binom_test(k, n, p):
     k or j are zero, only the non-zero tail is integrated.
     """
     mode = lbinom_pmf_mode(0, n, 0, (n, p))
-
-    if k <= mode:
-        j_min, j_max = mode, n
-    else:
-        j_min, j_max = 0, mode
-
+    distance = abs(mode - k)
     target = lbinom_pmf(k, n, p)
-    j = lbinom_pmf_inverse(j_min, j_max, target, (n, p))
-
+   
+    if k < mode: 
+        j_min = mode
+        j_max = min(mode + 1.5 * distance, n)
```

Comment:
why is j_max not just n? are there no cases where one crossing is more than 1.5 times the distance from the mode than the other one?

Reply:
Approaching high S1, the secant method has hard time to find the right j if we ask it to look between [0, mode] or [mode, n] since the interval is very big. Generally the relation:

- for k < mode: P(mode; n, p) < P(k; n, p) < P(mode + 1.5 * distance; n, p) 
- for k > mode: P(mode - 1.5 * distance; n, p) < P(k; n, p) < P(mode; n, p) 

is not true. Anyway, it is reasonably true with a tolerance of 1e-6. However, we could relax it a little, for instance to 2.
The only problem with finding j is when we are extremely close to the mode - especially for high S1, above 1e4. Here, no matter the range, the tolerance is worst. But I don't think it is a problem since we are far from the "bad" p-values.

Here a snip code for testing it with the current binomial test:
```
def valide_distance(f, n, p, step=0.001):
    print(f'n: {n:.6f}\tp: {p:.6f}')
    mode = lbinom_pmf_mode(0, n, 0, (n, p))
    print(f'mode: {mode}')
    k_check = np.arange(0, n, step)
    for k in tqdm.tqdm(k_check):
        target = lbinom_pmf(k, n, p)
        
        distance = abs(mode - k)
        if k != 0 and k != n and binom_pmf(k, n, p) > 1e-06:
            if k < mode: 
                    j_min = mode
                    j_max = min(mode + f * distance, n)
                    j = lbinom_pmf_inverse(j_min, j_max, target, (n, p))
                    if not np.isnan(j) and j != n:
                        if not binom_pmf(j_min, n, p) > binom_pmf(k, n, p) > binom_pmf(j_max, n, p):
                            print(j_min, k, j_max)
                            print(binom_pmf(j_min, n, p), binom_pmf(k, n, p), binom_pmf(j_max, n, p))
                            break
                        if not np.isclose(binom_pmf(j, n, p), binom_pmf(k, n, p), rtol=0, atol=1e-06, equal_nan=False):
                            print(j, k)
                            print(binom_pmf(j, n, p), binom_pmf(k, n, p))
                            break
            elif k > mode:
                    j_min = max(mode - f * distance, 0)
                    j_max = mode
                    j = lbinom_pmf_inverse(j_min, j_max, target, (n, p))
                    if not np.isnan(j) and j != n:
                        if not binom_pmf(j_min, n, p) < binom_pmf(k, n, p) < binom_pmf(j_max, n, p):
                            print(j_min, k, j_max)
                            print(binom_pmf(j_min, n, p), binom_pmf(k, n, p), binom_pmf(j_max, n, p))
                            break
                        if not np.isclose(binom_pmf(j, n, p), binom_pmf(k, n, p), rtol=0, atol=1e-06, equal_nan=False):
                            print(j, k)
                            print(binom_pmf(j, n, p), binom_pmf(k, n, p))
                            break
            else:
                pass
```

Reply:
For that high S1s, the gaussian approximation is already excellent-- should we use that, perhaps?

Reply:
Yes sure, do you want to me to do it or can we leave it for the next version? I can open an issue to remember that

Reply:
open issue is fine